File: java-samples/Accounts/supplychain/contracts/src/main/java/net/corda/samples/supplychain/contracts/InvoiceStateContract.java
package net.corda.samples.supplychain.contracts;

import net.corda.core.contracts.CommandData;
import net.corda.core.contracts.CommandWithParties;
import net.corda.core.contracts.Contract;
import net.corda.core.contracts.ContractState;
import net.corda.core.transactions.LedgerTransaction;
import net.corda.samples.supplychain.states.InvoiceState;

import java.util.List;

import static net.corda.core.contracts.ContractsDSL.requireSingleCommand;
import static net.corda.core.contracts.ContractsDSL.requireThat;

// ************
// * Contract *
// ************
public class InvoiceStateContract implements Contract {
    // This is used to identify our contract when building a transaction.
    public static final String ID = "net.corda.samples.supplychain.contracts.InvoiceStateContract";

    // A transaction is valid if the verify() function of the contract of all the transaction's input and output states
    // does not throw an exception.
    @Override
    public void verify(LedgerTransaction tx) throws IllegalArgumentException {
        /*
         * We can use the requireSingleCommand function to extract command data from transaction.
         * However, it is possible to have multiple commands in a signle transaction.
         */
        final CommandWithParties<InvoiceStateContract.Commands> command = requireSingleCommand(tx.getCommands(), InvoiceStateContract.Commands.class);

        List<ContractState> inputs = tx.getInputStates();
        List<ContractState> outputs = tx.getOutputStates();

        if (command.getValue() instanceof InvoiceStateContract.Commands.Create) {

            // Using Corda DSL function requireThat to replicate conditions-checks
            requireThat(require -> {
                require.using("No inputs should be consumed when creating a new Invoice State.", inputs.isEmpty());
                require.using("Transaction must have exactly one output.", outputs.size() == 1);
                InvoiceState output = (InvoiceState) outputs.get(0);
                require.using("Invoice amount must be a valid number (Greater than zero)", output.getAmount() > 0);
                return null;
            });

        } else {
            throw new IllegalArgumentException("Command not found!");
        }

    }

    // Used to indicate the transaction's intent.
    public interface Commands extends CommandData {
        class Create implements Commands {}
    }
}

File: kotlin-samples/Accounts/sharestatewithaccount/contracts/src/main/kotlin/net/corda/samples/supplychain/contracts/InvoiceStateContract.kt
package net.corda.samples.supplychain.contracts


import net.corda.core.contracts.*
import net.corda.core.transactions.LedgerTransaction
import net.corda.samples.supplychain.states.InvoiceState

class InvoiceStateContract : Contract{

    companion object{
        const val ID = "net.corda.samples.supplychain.contracts.InvoiceStateContract"
    }

    override fun verify(tx: LedgerTransaction) { /*
         * We can use the requireSingleCommand function to extract command data from transaction.
         * However, it is possible to have multiple commands in a signle transaction.
         */
        val command = tx.commands.requireSingleCommand<Commands>()
        when (command.value) {
            is Commands.Create -> requireThat {
                val inputs = tx.inputStates
                val outputs = tx.outputStates

                // no inputs to the transaction
                "No inputs should be consumed when creating a new SantaSession.".using(inputs.isEmpty())
                "Transaction must have exactly one output.".using(outputs.size == 1)

                val output = outputs[0] as InvoiceState
                // must be three or more players
                "Invoice amount must be a valid number (Greater than zero)".using(output.amount > 0)
                null
            }
        }
    }

    interface Commands : CommandData {
        class Create : Commands
    }
}